NewsApp/Services/CoreData/CoreDataManager.swift

```diff
@@ -0,0 +1,172 @@
+//
+//  CoreDataManager.swift
+//  NewsApp
+//
+//  Created by Abdelrhman Eliwa on 11/08/2021.
+//
+
+import UIKit
+import CoreData
+
+class CoreDataManager {
+    // MARK: - Properties
+    private let defaults = UserDefaults.standard
+    private var managedContext: NSManagedObjectContext? {
```

Comment:
Build in init and return nullable init if failed to be create to avoid multiple nil check